XenAdmin/Diagnostics/Checks/PatchPrecheckCheck.cs

```diff
@@ -71,6 +73,14 @@ public PatchPrecheckCheck(Host host, Pool_patch patch, Dictionary<string, livepa
             this.livePatchCodesByHost = livePatchCodesByHost;
         }
 
+        public PatchPrecheckCheck(Host host, Pool_update update, Dictionary<string, livepatch_status> livePatchCodesByHost, List<SR> srUploadedUpdates)
+            : base(host)

```

Comment:
Here can use existing `PatchPrecheckCheck` constructor to avoid code repetition (i.e. use `this` instead of `base`):
```
public PatchPrecheckCheck(Host host, Pool_update update, Dictionary<string, livepatch_status> livePatchCodesByHost, List<SR> srUploadedUpdates)
      : this(host, update, livePatchCodesByHost)
{
    this.srUploadedUpdates = srUploadedUpdates;
}
```